File: src/main/Java/boj_step9/boj5086.java
import java.io.*;
import java.util.StringTokenizer;

public class boj5086 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        while (true) {
            StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
            int a = Integer.parseInt(tokenizer.nextToken());
            int b = Integer.parseInt(tokenizer.nextToken());
            if (a == 0 && b == 0) break;
            if (a % b == 0) writer.append("multiple");
            else if (b % a == 0) writer.append("factor");
            else writer.append("neither");
            writer.newLine();
        }
        writer.close();
    }
}

File: src/main/Kotlin/boj_step9/boj5086.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w =  BufferedWriter(OutputStreamWriter(System.out))
    while(true) {
        val (a, b) = r.readLine().split(" ").map { it.toInt() }
        if(a == 0 && b == 0) break
        val result = if(b % a == 0) "factor\n" else if (a % b == 0) "multiple\n" else "neither\n"
        w.write(result)
    }
    w.close()
}